tests/test_skills/test_abstract_round_abci/test_behaviours_utils.py

```diff
@@ -569,11 +569,75 @@ def test_get_request_nonce_from_dialogue(self) -> None:
         result = BaseBehaviour._get_request_nonce_from_dialogue(dialogue_mock)
         assert result == expected_value
 
+    @mock.patch.object(BaseBehaviour, "_send_signing_request")
+    @mock.patch.object(Transaction, "encode", return_value=MagicMock())
+    @mock.patch.object(
+        BaseBehaviour,
+        "_build_http_request_message",
+        return_value=(MagicMock(), MagicMock()),
+    )
+    @mock.patch.object(BaseBehaviour, "_check_http_return_code_200", return_value=False)
+    def test_send_transaction_stop_condition(self, *_: Any) -> None:
+        """Test '_send_transaction' method's `stop_condition` as provided by `send_a2a_transaction`."""
+        # create the exact same stop condition that we create in the `send_a2a_transaction` method
+        stop_condition = self.behaviour.is_round_ended(
+            self.behaviour.matching_round.round_id
+        )
+        gen = self.behaviour._send_transaction(
+            MagicMock(),
+            request_retry_delay=0,
+            stop_condition=stop_condition,
+        )
+        # assert that the stop condition does not apply yet
+        assert not stop_condition()
+        # trigger the generator function so that we enter the `stop_condition` loop
+        try_send(gen)
+
+        # set the current round's id so that it meets the requirements for a `stop_condition`
+        self.behaviour.context.skill_context.state.round_sequence.current_round_id = (
+            self.behaviour.matching_round.round_id
+        )
+        # assert that everything was set as expected
+        assert (
+            self.behaviour.context.skill_context.state.round_sequence.current_round_id
+            == self.behaviour.matching_round.round_id
+            == "round_a"
+        )
+        # assert that the stop condition now applies
+        assert stop_condition()
```

Comment:
Interesting...I think it might be explained by the generator not having been triggered again?

Reply:
I think not, because moving the assertion below [line 626](https://github.com/valory-xyz/open-autonomy/pull/980/files#diff-63b131ca6614d6666a3eeb925cb9b9e71b63f27a68ea748d9db7f8bae21f0bc5R626 ) does not help.